pkg/tfvars/vsphere/vsphere.go

```diff
@@ -100,3 +114,84 @@ func convertVCentersToMap(values []vtypes.VCenter) map[string]vtypes.VCenter {
 	}
 	return vcenterMap
 }
+
+func constructKargsFromNetworkConfig(ipAddrs, nameservers []string, gateway4, gateway6, deviceName string) (string, error) {
+	if logrus.IsLevelEnabled(logrus.TraceLevel) {
+		logrus.Tracef("Constructing kargs from IPs [%v] nameservers [%v] gateway4 [%v]", ipAddrs, nameservers, gateway4)
+	}
+	outKargs := ""
+	// if an IPv4 gateway is defined, we'll only handle IPv4 addresses
+	for _, address := range ipAddrs {
+		ip, mask, err := net.ParseCIDR(address)
+		if err != nil {
+			return "", err
+		}
+
+		var gatewayStr string
+		var maskStr string
+		var ipStr string
+		if ip.To4() == nil {
+			gatewayStr = gateway6
+			if len(gatewayStr) > 0 {
+				gatewayStr = fmt.Sprintf("[%s]", gatewayStr)
+			}
+			maskStr = mask.String()
```

Comment:
There isn't a built in method for this in ipv6?

Reply:
the wrapping of the IP address in brackets, I think is specific to dracut.  this entire portion of code was refactored recently though to take more advantage of existing libraries.